rust/template/differential_datalog/program.rs

```diff
@@ -1471,8 +1504,24 @@ impl<V: Val> Program<V> {
     }
 
     /* Returns all input relations of the program */
-    fn input_relations(&self) -> Vec<RelId> {
+    fn input_relations(&self) -> Response<Vec<RelId>> {
```

Comment:
I think this one is a bit too much on the paranoid side. I suggest we use `assert!()` instead of returning the error and keep the return type of this function as `Vec<RelId>`.